src/views/upload/components/upload.js

```diff
@@ -1,57 +1,228 @@
 import React, { PropTypes, Component } from 'react';
+import { Button, FormControl, Col, ControlLabel, Checkbox,
+} from 'react-bootstrap/lib/';
+import TagsInput from 'react-tagsinput';
 import './upload.scss';
 
+
 export default class Upload extends Component {
 
     static propTypes = {
+        groups: PropTypes.arrayOf(Object),
         updateUploadFile: PropTypes.func.isRequired,
         submitFile: PropTypes.func.isRequired,
+        fetchItemTypes: PropTypes.func.isRequired,
+        setAllItemID: PropTypes.func.isRequired,
+        fieldVisible: PropTypes.boolean,
+        tags: PropTypes.arrayOf(String),
+        handleTagsChange: PropTypes.func.isRequired,
+        allItemID: PropTypes.arrayOf(String),
+        filePicked: PropTypes.boolean,
+        setFieldVisible: PropTypes.func.isRequired,
+        setFilePicked: PropTypes.func.isRequired,
+        resetFile: PropTypes.func.isRequired,
+
+        // holds all values from meta data text fields
+        allMetaDataValue: PropTypes.arrayOf(Object),
+        setAllMetaData: PropTypes.func.isRequired,
     };
 
-    static renderMetadataInput(fields) {
-        return fields.map((field, id) =>
-            (
-                <div className='upload-metadata-input' key={id}>
-                    <span className='upload-label'>{field.label}</span>
-                    <input type={field.type} />
-                </div>
-            )
-        );
+    componentWillMount() {
+        const { fetchItemTypes, setAllItemID, setAllMetaData, setFieldVisible, handleTagsChange, resetFile, setFilePicked } = this.props;
+        fetchItemTypes();
+        const metaData = [];
+        const allItemID = [];
+        const allTags = [];
+        setAllMetaData(metaData);
+        setAllItemID(allItemID);
+        handleTagsChange(allTags);
+        setFieldVisible(false);
+        setFilePicked(false);
+        resetFile();
+    }
+
+    handleSubmit = () => {
+        const { submitFile, tags, allMetaDataValue, filePicked } = this.props;
+        let metaDataArray;
+        if (allMetaDataValue !== undefined) {
+            metaDataArray = allMetaDataValue.slice();
+        } else {
+            metaDataArray = [];
+        }
+        if (filePicked === false) {
+            alert('Please click browse to select a file to upload');
+        } else if (metaDataArray.length <= 0) {
+            alert('Please pick a category and enter some meta data for this file before uploading');
+        } else {
+            for (let i = 0; i < metaDataArray.length; i += 1) {
+                console.log(metaDataArray[i].data);
+            }
+            submitFile(tags, allMetaDataValue);
+            // reset the page after done uploading
+            this.componentWillMount();
```

Comment:
you shouldn't be calling this directly. instead you should make a function called something like `resetPage = () => {..}` that calls all of the reset functions. then update `componentWillMount` to just call `this.reset()`